src/hex/coordinate.rs

```diff
@@ -202,6 +223,153 @@ impl Axial {
         self.make_vector(1, direction.into())
     }
 
+    /// Get all the neighbors for this coordinate.
+    ///
+    /// See [`HexDirection`] for a reference of directionality.
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial};
+    ///
+    /// let coord = axial!(1, 2);
+    ///
+    /// if axial!(0, 0).neighbors().contains(&coord) {
+    ///     // Do something ...
+    /// }
+    /// ```
+    pub fn neighbors(&self) -> [Self; 6] {
+        [
+            self.neighbor(HexDirection::Front),
+            self.neighbor(HexDirection::FrontRight),
+            self.neighbor(HexDirection::BackRight),
+            self.neighbor(HexDirection::Back),
+            self.neighbor(HexDirection::BackLeft),
+            self.neighbor(HexDirection::FrontLeft),
+        ]
+    }
+
+    /// Checks if ALL of the supplied coordinates are neighbors to self.
+    ///
+    /// See [`Axial::neighbor`].
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial};
+    ///
+    /// let coord = axial!(1, 2);
+    ///
+    /// if axial!(0, 0).are_neighbors(&[coord]) {
+    ///     // Do something ...
+    /// }
+    /// ```
+    pub fn are_neighbors(&self, coords: &[Self]) -> bool {
+        for coord in coords {
```

Comment:
i feel fairly certain this can be optimised a bit (eg: the call to `neighbors()` could be made up to 6 times in a reasonable use case)